compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -509,12 +509,20 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
                         if let (ty::ConstKind::Unevaluated(a), ty::ConstKind::Unevaluated(b)) =
                             (c1.kind(), c2.kind())
                         {
-                            if infcx.try_unify_abstract_consts(
-                                a.shrink(),
-                                b.shrink(),
-                                obligation.param_env,
+                            if let (Ok(Some(ct_a)), Ok(Some(ct_b))) = (
+                                tcx.bound_abstract_const(a.shrink()),
+                                tcx.bound_abstract_const(b.shrink()),
                             ) {
-                                return ProcessResult::Changed(vec![]);
+                                let a = ct_a.subst(tcx, a.substs);
+                                let b = ct_b.subst(tcx, b.substs);
+
+                                if infcx
+                                    .at(&obligation.cause, obligation.param_env)
+                                    .eq(a, b)
+                                    .is_ok()
```

Comment:
you need to deal with the nested obligations here and add then to the `ProcessResult`s

Reply:
If I change this, it then tries to equate an `Const::Expr` with what looks like a constant + integer, is that something that should be fixed in this file or elsewhere?

Reply:
i don't fully get what you mean here? can you elaborate a bit, either here or in zulip?